### PR TITLE
New CSS Cheat Sheet

### PR DESCRIPTION
**Pull request by ethanchewy**
## Demo

Demo GIF

Video Demo link: https://drive.google.com/file/d/0B_0VuTZLjvTQalkzXy1FYkY5T0k/view?usp=sharing 
**What does your Instant Answer do?**
It allows users to quickly reference CSS syntax styling and properties. I have incorporated some of the most popular and most used properties. 

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It allows users to quickly see all essential CSS properties and styles in a streamline display. No link or cheatsheet link really aggregates all of what I have listed. For example, a user has to go to a a lot of different links on w3schools to understand these CSS properties. This cheatsheet displays all that information succinctly. 

**What is the data source for your Instant Answer? (Provide a link if possible)**
I looked up these properties from w3schools (http://www.w3schools.com/css/). I structured it similarly to http://www.lesliefranke.com/files/reference/csscheatsheet.html

**Why did you choose this data source?**
W3schools is a benchmark and reputable source for css. It references mozilla developer css information and creates its own explanations for css. Throughout the programming community, w3school is one of the most accessible and accurate css resource. It is also very succinct.

**Are there any other alternative (better) data sources?**
Not really. Mozilla's CSS resource is as good as the w3school (https://developer.mozilla.org/en-US/docs/Web/CSS). It's a little more complicated and goes more in depth. 

**What are some example queries that trigger this Instant Answer?**
"css cheatsheet", "style sheet cheatsheet" (Note: I will be adding more example queries later on. Just wanted to create this basic pull request first.)

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Web Developers, Designers, etc.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes, I contacted Zaahir and he said that even though it was claimed, I can go ahead and make it.

**Which existing Instant Answers will this one supercede/overlap with?**
No one has created this yet even though it was claimed a while ago.

**Are you having any problems? Do you need our help with anything?**
Not really. I overcame the JSON syntax errors pretty quickly.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I learned it from my other developer friend a few years ago. I have been using it ever since.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

## Checklist
Please place an 'X' where appropriate.

- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 10
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] IE 11
        - [X] Microsoft edge
        
    - Windows 8
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] IE 10

    - Windows 7
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] IE 8
        - [X] IE 9
        - [X] IE 10

    - Windows XP
        - [X] IE 7
        - [X] IE 8

    - Linux
        - [X] Google Chrome
        - [X] Firefox
        
    - Mac OSX
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] Safari

    - iOS (iPhone)
        - [X] Safari Mobile
        - [X] Google Chrome
        - [X] Opera

    - iOS (iPad)
        - [X] Safari Mobile
        - [X] Google Chrome
        - [X] Opera

    - Android
        - [X] Firefox
        - [X] Native Browser
        - [X] Google Chrome
        - [X] Opera

-----
IA Page: https://duck.co/ia/view/css_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@ethanchewy Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

File: share/goodie/cheat_sheets/json/css.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
+{
+    "id": "css_cheat_sheet",
+    "name": "CSS",
+    "description": "Explanation of CSS Syntax and Properties",
+    "metadata": {
+        "sourceName": "w3schools",
+        "sourceUrl": "http://www.w3schools.com/css/default.asp"
+    },
+    "aliases": [
+        "css", "cascading style sheets", "style sheet"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/css.json]: the `css` alias isn't necessary as the filename `css.json` (minus the `.json`) is the default trigger.

I would suggest changing this alias to `css language` perhaps? Or simply remove it.

**Comment by moollaza** [comment]
@ethanchewy this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by ethanchewy** [comment]
@moollaza Thanks! I'll definitely edit this json file so that there are more triggers. I'm going to help out the python dev with his cheatsheet (probably create a different pull request).

**Comment by moollaza** [comment]
@ethanchewy congrats! It's live: https://duckduckgo.com/?q=CSS+Cheat+Sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**